zzk/service/hostregistry.go

```diff
@@ -250,8 +251,21 @@ func (h *HostRegistryListener) Spawn(cancel <-chan interface{}, hostid string) {
 						// soon as possible.
 						select {
 						case <-h.isOnline:
-							count := DeleteHostStates(h.conn, h.poolid, hostid)
+							// Only reschedule services without address
+							// assignments.
+							count := DeleteHostStatesWhen(h.conn, h.poolid, hostid, func(s *State) bool {
+								return s.DesiredState == service.SVCStop || s.AssignedIP == ""
```

Comment:
That is a good point.  Let me see if I can figure out a way to handle virtual ips.